### PR TITLE
heartbeat backward comptability

### LINKED ISSUE #4418: Compute nodes 1.4 can't join 1.5 orchestrator
1.4 compute nodes are always in disconnected state

### Steps to reproduce
1. Start v1.5.0 orchestrator:
    
    ```bash
    # start with fresh data dir
    export BACALHAU_DIR=$(mktemp -d)
    bacalhau-1.5.0 serve --orchestrator 
    
    ```
    
2. Start v1.4.0 compute node:
    
    ```bash
    # start with fresh data dir
    export BACALHAU_DIR=$(mktemp -d)
    # listen to a different API port as default one is taken by the orchestrator
    export BACALHAU_NODE_SERVERAPI_PORT=1235
    bacalhau-1.4.0 serve --node-type compute --orchestrators 127.0.0.1:4222
    ```
    
3. List nodes 
    
    ```bash
    export BACALHAU_DIR=~/bacalhau-versions/v1.5.0/data-dir
    bacalhau-1.5.0 node list


     ID          TYPE       APPROVAL  STATUS        LABELS                                      CPU     MEMORY      DISK         GPU
     n-71e962df  Requester  APPROVED                Architecture=arm64 Operating-System=darwin
     n-738e97d7  Compute    APPROVED  DISCONNECTED  Architecture=arm64 Operating-System=darwin  6.4 /   12.8 GB /   43.1 GB /    0 /
                                                                                            6.4     12.8 GB     43.1 GB      0

    ```

### LINKED ISSUE #4419: Compute nodes 1.5 can't join 1.4 orchestrator
1.5 compute nodes are always in disconnected state

### Steps to reproduce
1. Start v1.4.0 orchestrator:
    
    ```bash
    # start with fresh data dir
    export BACALHAU_DIR=$(mktemp -d)
    bacalhau-1.4.0 serve --node-type requester
    
    ```
    
2. Start v1.5.0 compute node:
    
    ```bash
    export BACALHAU_DIR=$(mktemp -d)
    bacalhau-1.5.0 serve --compute -c api.port=1235
    ```
    
3. List nodes 
    
    ```bash
    export BACALHAU_DIR=~/bacalhau-versions/v1.4.0/data-dir
    bacalhau-1.4.0 --api-host 127.0.0.1 node list

     ID          TYPE       APPROVAL  STATUS        LABELS                                      CPU     MEMORY      DISK         GPU
     n-22b3497b  Requester  APPROVED                Architecture=arm64 Operating-System=darwin
     n-93469683  Compute    APPROVED  DISCONNECTED  Architecture=arm64 Operating-System=darwin  5.6 /   11.2 GB /   37.6 GB /    0 /
                                                                                            5.6     11.2 GB     37.6 GB      0

    ```
    
In addition to that, you can see errors in the orchestrator's logs:
```
10:05:25.961 | ERR pkg/nats/pubsub/pubsub.go:93 > error unmarshalling nats payload from subject heartbeat error="invalid character '\\x01' looking for beginning of value"
10:05:40.959 | ERR pkg/nats/pubsub/pubsub.go:93 > error unmarshalling nats payload from subject heartbeat error="invalid character '\\x01' looking for beginning of value"
```

This is related to using new `ncl` library to heartbeat in 1.5

### PR DESCRIPTION
**Pull request by wdbaruni**
Nodes in v1.5.0 were still heartbeating to the same existing `heartbeat` topic as nodes `v1.4.0`, but they are serializing messages differently using ncl library. This causes nodes to show as `DISCONNECTED` if the orchestrator and compute nodes are using different versions as they fail to deserialize the heartbeat.

This PR fixes this by:
1. Using a new topic for heartbeats in v1.5.0 that follows the pattern documented in [Reliable Orchestration Design doc](https://www.notion.so/expanso/Reliable-Orchestration-Design-397083a957794668adb15322553e6652?pvs=4), which follows `bacalhau.global.compute.<node_id>.out.heartbeat`. This allows replacing global with different regions in the future, and allows restricting nodes to heartbeat only to their dedicated topics
2. Compute nodes in v1.5.0 will heartbeat to both new and old topics to allow them to join v1.4.0 networks
3. Orchestrator node in v1.5.0 will subscribe to both heartbeat topics to allow v1.4.0 compute nodes to join its network

### Testing
Verified this scenario is passing
https://www.notion.so/expanso/Cross-Version-Compatibility-Tests-e8031b5ed7554b28abda5fe4069fc363?pvs=4#f069d285b5824e599cd1fb2c00dc5785

Closes #4418 
Closes #4419

### CODE REVIEW

File: pkg/config_legacy/configenv/production.go [hunk 1]
```diff
@@ -125,7 +125,6 @@ var ProductionComputeConfig = types.ComputeConfig{
 		InfoUpdateFrequency:     types.Duration(60 * time.Second),
 		ResourceUpdateFrequency: types.Duration(30 * time.Second),
 		HeartbeatFrequency:      types.Duration(15 * time.Second),
-		HeartbeatTopic:          "heartbeat",
```
Comment (frrist) [inline pkg/config_legacy/configenv/production.go:128]: Let's revert all changes made to config_legacy package.

**Comment by wdbaruni** [comment]
Yes related to the mentioned PR. The fix is explained in the PR's description

File: pkg/compute/management_client.go [hunk 2]
```diff
@@ -158,6 +158,9 @@ func (m *ManagementClient) Start(ctx context.Context) {
 		heartbeatTicker.Stop()
 		resourceTicker.Stop()
 		infoTicker.Stop()
+
+		// Close the heartbeat client and it's resources
+		m.heartbeatClient.Close(ctx)
```
Comment (frrist) [inline pkg/compute/management_client.go:163]: :+1:

File: pkg/node/ncl.go [hunk 3]
```diff
@@ -15,3 +16,16 @@ func CreateMessageSerDeRegistry() (*ncl.MessageSerDeRegistry, error) {
...
+// the orchestrator subscribes to.
+func computeHeartbeatTopic(nodeID string) string {
+	return fmt.Sprintf("bacalhau.global.compute.%s.out.heartbeat", nodeID)
+}
+
+// orchestratorHeartbeatSubscription returns the subject to subscribe for compute heartbeats.
+// it subscribes for heartbeat messages from all compute nodes
+func orchestratorHeartbeatSubscription() string {
+	return "bacalhau.global.compute.*.out.heartbeat"
+}
```
Comment (frrist) [inline pkg/node/ncl.go:31]: nit: might be better to extract `bacalhau.global.compute.%s.out.heartbeat` to a variable e.g.
```suggestion
const HeartbeatTopicFormat = "bacalhau.global.compute.%s.out.heartbeat"

// computeHeartbeatTopic returns the subject to publish heartbeat messages to.
// it publishes to the outgoing heartbeat subject of a specific compute node, which
// the orchestrator subscribes to.
func computeHeartbeatTopic(nodeID string) string {
	return fmt.Sprintf(HeartbeatTopicFormat, nodeID)
}

// orchestratorHeartbeatSubscription returns the subject to subscribe for compute heartbeats.
// it subscribes for heartbeat messages from all compute nodes
func orchestratorHeartbeatSubscription() string {
	return fmt.Sprintf(HeartbeatTopicFormat, "*")
}
```

File: pkg/config_legacy/configenv/production.go (hunk 1 above)
Comment (wdbaruni) [reply pkg/config_legacy/configenv/production.go]: Done

**Pull request merged**